nvm.sh

```diff
@@ -2443,30 +2442,29 @@ nvm() {
 
         # skip binary install if "nobinary" option specified.
         if [ $nobinary -ne 1 ] && nvm_binary_available "$VERSION"; then
-          if nvm_install_binary "${FLAVOR}" std "${VERSION}"; then
-            NVM_INSTALL_SUCCESS=true
-          fi
+          nvm_install_binary "${FLAVOR}" std "${VERSION}"
+          EXIT_CODE=$?
         fi
-        if [ "$NVM_INSTALL_SUCCESS" != true ]; then
+        if [ "$EXIT_CODE" -ne 0 ]; then
```

Comment:
if we're going to do this check, we should set it to `-1` initially, or else `set -u` might complain.